File: C++/stack.cpp
#include<iostream>
#include<string>
using namespace std;
class stack
{
   private:
   int top;
   int arr[5];
    
public:
    stack()
    {
        top=-1;
        for(int i=0;i<5;i++)
        {
            arr[i]=0;
        }

    }

    bool isEmpty()
    {
       if(top==-1)
           return true;
       else
           return false;
           
    }
    bool isFull()
    {
        if(top==4)
           return true;
        else 
           return false;

    }
    void push(int val) //yes argument no return 
    {
       if(isFull())
       {
           cout<<"Stack Overflow";
       }
       else
       {
          top++;
          arr[top]=val;
       }
       

    }
    int pop() //no argument yes return;
    {
        if(isEmpty())
        {
            cout<<"Stack Underflow";
            return 0;
        }
        else
        {
            int popValue=arr[top];
            arr[top]=0;
            top--;
            return popValue;
        }
        
    }

    void count()
    {
         cout<<top+1;
    }
    int peek(int pos)
    {
        if(isEmpty())
        {
            cout<<"Stack Underflow";
        }
        else
        {
            return arr[pos];
        }
        
    }
    void change(int pos,int val)
    {
        arr[pos]=val;
        
    }
    void display()
    {
        for(int i=4;i>=0;i--)
        {
            cout<<arr[i]<<endl;
        }
    }
    








};
int main()
{   stack s1;
    s1.push(5);
    s1.push(3);
    s1.push(-1);
    s1.push(2);
    s1.pop();
    s1.pop();
  s1.display();
  s1.count();


    
    
    

    return 0;

}

File: Java/stack.java
import java.io.*;
import java.util.Scanner;
class stack
{
private int arr[];
private int  top;
private int size;
stack(int s)
{

	arr=new int[s];
	size=s;
	top=-1;
}
public void push(int num)
{
if(isFull())
{
System.out.println("Stack Overflow");
System.exit(1);
}
System.out.println("Inserting "+num);
arr[++top]=num;
 	
 	}
public int pop()
{
	if(isEmpty())
	{
	System.out.println("Stack is underflow!!!");
	System.exit(1);
	}
	System.out.println("Removing "+peek());
	return arr[top--];

}
public int peek()
{
	if(!isEmpty())
	  return arr[top];
	else 
	 System.exit(1);

	return -1;
}
public int s()
{
	return top+1;
}
public boolean isEmpty()
{
	return top==-1;
}
public boolean isFull()
{
	return top==size-1;
}
public static void main(String[] args)

{
 Scanner myObj = new Scanner(System.in);
 System.out.println("Enter the number of elements!!");
 int i=myObj.nextInt();
 stack S=new stack(i);

 for(int j=0;j<i;j++)
 {
  int myInt=myObj.nextInt();
  S.push(myInt);

 }
 S.pop();

	
}
}